hw3/bleichenbacher.py

```diff
@@ -120,7 +144,18 @@ def search_single_interval(k, key, B, prev_s, a, b, c_0, oracle):
     :param oracle: oracle that checks ciphertext conformity
     :return: s s.t. (c_0 * (s ** e)) mod n represents a conforming ciphertext
     """
-    ?
+    r_lower_bound = divceil(2 * ((b * prev_s) - (2 * B)), key.n)
+    next_r = r_lower_bound
+    while True:
+        s_lower_bound = divceil((2 * B) + (next_r * key.n), b)
+        s_upper_bound = divceil((3 * B) + (next_r * key.n), a)

```

Comment:
Not sure about the wording, but I just got confused again about why this is ceil and this is the explanation.
```suggestion
        # Use `divceil` in order to get integers.
        # For the upper bound, we use ceil (and not floor) as we enumerate until the last integer
        # smaller than `s_upper_bound`, and we want the floor value to be included in this range.
        s_lower_bound = divceil((2 * B) + (next_r * key.n), b)
        s_upper_bound = divceil((3 * B) + (next_r * key.n), a)

```